pwiz_tools/Skyline/TestUtil/HttpClientTestHelper.cs

```diff
@@ -1016,9 +1223,131 @@ internal void CompleteSuccess(RecordingEntry entry)
 
         private static void AssignResponseBody(HttpInteraction interaction, byte[] bytes)
         {
-            interaction.ResponseBody = bytes == null || bytes.Length == 0
-                ? string.Empty
-                : Encoding.UTF8.GetString(bytes);
+            if (bytes == null || bytes.Length == 0)
+            {
+                interaction.ResponseBody = string.Empty;
+                interaction.ResponseBodyIsBase64 = false;
+                return;
+            }
+
+            // Check if content is binary based on ContentType
+            bool isBinary = IsBinaryContentType(interaction.ContentType);
+            
+            // If not determined by ContentType, try to decode as UTF-8 to detect binary
+            if (!isBinary)
+            {
+                try
+                {
+                    var testString = Encoding.UTF8.GetString(bytes);
+                    // Verify round-trip: if re-encoding doesn't match, it's likely binary
+                    var reencoded = Encoding.UTF8.GetBytes(testString);
+                    isBinary = !bytes.SequenceEqual(reencoded);
+                }
+                catch
+                {
+                    isBinary = true; // If decoding fails, treat as binary
+                }
+            }
+
+            if (isBinary)
+            {
+                // Encode binary content as base64 and split into lines for better JSON readability
+                // (similar to how .NET .resx files format base64 content)
+                var base64String = Convert.ToBase64String(bytes);
+                interaction.ResponseBody = base64String;
+                interaction.ResponseBodyIsBase64 = true;
+                // Split base64 into lines (76 characters per line is a common standard)
+                interaction.ResponseBodyLines = SplitBase64IntoLines(base64String);
+            }
+            else
+            {
+                // Store text content as UTF-8 string
+                var text = Encoding.UTF8.GetString(bytes);
+                interaction.ResponseBody = text;
+                interaction.ResponseBodyIsBase64 = false;
+                
+                // If text contains newlines, also store as array of lines for better JSON readability
+                if (text.Contains("\n") || text.Contains("\r\n"))
```

Comment:
The `text.Contains("\r\n")` is redundant since you already checked `text.Contains("\n")`